feat: Ledger sing message support

JordiParraCrespo opened the pull request:
# Ledger sing message support

This PR adds support for message signing using Ledger hardware wallet.

**Key Changes:**

- Added `signMessage` and `signDelegateAction` methods to the `ledger-client`.
- Optimized `CHUNK_SIZE` to reduce the number of communication calls to the Ledger device.
- Integrated `signMessage` into the` ledger` class.
- Added unit tests to cover the new message signing functionality. 

# Checklist:
<!-- CHECKLIST_TYPE: ALL -->
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
<!-- /CHECKLIST_TYPE -->

# Type of change.
<!-- CHECKLIST_TYPE: ONE -->
- [ ] FIX - a PR of this type patches a bug.
- [x] FEATURE - a PR of this type introduces a new feature.
- [ ] BUILD - a PR of this type introduces build changes.
- [ ] CI - a PR of this type introduces CI changes.
- [ ] DOCS - a PR of this type introduces DOCS improvement.
- [ ] STYLE - a PR of this type introduces style changes.
- [ ] REFACTOR - a PR of this type introduces refactoring.
- [ ] PERFORMANCE - a PR of this type introduces performance changes.
- [ ] TEST - a PR of this type adds more tests.
- [ ] CHORE - a PR introduces other changes than the specified above.
<!-- /CHECKLIST_TYPE -->

Commit:
Merge pull request #1204 from near/dev

v8.9.12 Release (dev → main)

gtsonevv committed:
Merge pull request #1220 from near/dev

v8.9.14 Release (dev → main)

kujtimprenku committed:
Remove @near-js/ aliasis from tsconfig.base.json.

kujtimprenku committed:
Remove @near-js import and use the near-api-js directly.

kujtimprenku committed:
Remove un-necessary peer dependency @near-js/providers.

kujtimprenku committed:
Bump borsh to v1.0.0 and remove the better-sqlite3

thisisjoshford committed:
feat: add to devtools projet workflow

JordiParraCrespo committed:
feat(ledger): add sign message to ledger

JordiParraCrespo committed:
test(ledger): update test for ledger and ledger-client with sign message

JordiParraCrespo committed:
fix(ledger): improve error message

JordiParraCrespo committed:
style(ledger): lint the code

amillez commented on packages/ledger/src/lib/ledger-client.spec.ts:
```diff
@@ -47,6 +47,19 @@ const createTransactionMock = () => {
+const createSignMessageMock = () => {
+  /**
+   * This is a hex encoded payload that is sent to the Ledger device.
+   * message: "Makes it possible to authenticate users without having to add new access keys. This will improve UX, save money and will not increase the on-chain storage of the users' accounts./Makes it possible to authenticate users without having to add new access keys. This will improve UX, save money and will not increase the on-chain storage of the users' accounts./Makes it possible to authe ...
```
Es repeteix 3 cops el mateix, no?

JordiParraCrespo replied:
Sii es okey. L,he tret del repo del la [app](https://github.com/LedgerHQ/app-near/blob/develop/tests/test_sign_nep413_msg/test_nep413_msg.py)

amillez commented on packages/ledger/src/lib/ledger.spec.ts:
```diff
@@ -174,6 +189,40 @@ describe("signAndSendTransactions", () => {
+    });
+
+    const message =
+      "Makes it possible to authenticate users without having to add new access keys. This will improve UX, save money and will not increase the on-chain storage of the users' accounts./Makes it possible to authenticate users without having to add new access keys. This will improve UX, save money and will not increase the on-chain storage of the users' accounts./Makes it possible to authenticate  ...
```
Idem

JordiParraCrespo replied:
Es okey!

amillez commented on packages/ledger/src/lib/ledger.ts:
```diff
@@ -276,6 +283,79 @@ const Ledger: WalletBehaviourFactory<HardwareWallet> = async ({
+      recipient,
+      callbackUrl,
+    }: SignMessageParams): Promise<SignedMessage | void> {
+      logger.log("signMessage", { message, nonce, recipient, callbackUrl });
```
Aquest log ha de ser-hi?

JordiParraCrespo replied:
x mi no xo ho fan als altres methods...

amillez reviewed (approved):
Nice job!!!🐐

gtsonevv committed:
Merge pull request #1229 from kujtimprenku/fix/mintbase-js-and-naj

fix: Remove @near-js/ aliases from tsconfig.base.json

gtsonevv committed:
Merge pull request #1230 from kujtimprenku/fix/remove-near-js-individual-packages

fix: Fix imports and remove unnecessary peer dependency

gtsonevv committed:
Merge pull request #1231 from kujtimprenku/feat/bump-borsh-dependency

feat: Bump borsh to v1.0.0 and remove the better-sqlite3

gtsonevv committed:
chore-bump-version-v8.9.15

gtsonevv committed:
Merge pull request #1268 from near/chore-bump-version-v8.9.15

chore-bump-version-v8.9.15

gtsonevv committed:
Merge pull request #1269 from near/dev

v8.9.15 Release (dev → main)

JordiParraCrespo committed:
Merge branch 'dev' of github.com:near/wallet-selector into HEAD

JordiParraCrespo committed:
fix(ledger): update to new borsh version

denbite committed:
fix: define range of `near-api-js` versions as peerDependency instead of an exact one

Elabar committed:
feat: meteor wallet in app selector

Elabar committed:
chore: update readme

Elabar committed:
chore: code linting

denbite committed:
Merge pull request #1272 from near/update_peer_dependencies

fix: define range of `near-api-js` versions as peerDependency instead of an exact one

Elabar committed:
feat: use nonce instead of incremental promise id

- also added a ping in signIn that should resolve almost immediately to avoid auto login stuck

Elabar committed:
feat: allow post message for all sites, add a source tag in message

- this is so our app make sure we are processing the right message

Elabar committed:
feat: pass window location href

trechriron committed:
Merge pull request #1275 from Elabar/meteor-wallet-mobile-app

feat: add support for meteor wallet mobile app

JordiParraCrespo committed:
Merge branch 'dev' of github.com:near/wallet-selector into ledger/feat/sign-message
